### PR TITLE
Optimize Active Record queries

### PR DESCRIPTION
**Pull request by jgarber623-gov**
## Checklist

I have…

- [x] run the application locally (`bin/rails server`) and verified that my changes behave as expected.
- [x] run static code analysis (`bin/rubocop`) and vulnerability scan (`bin/brakeman`) against my changes.
- [x] run the test suite (`bin/rake spec`) and verified that all tests pass.
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] attached screenshots illustrating relevant behavior before and after my changes.

## Summary of Changes

This pull request optimizes a few Active Record queries by taking advantage of [AR's eager loading features](http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations), refactoring pieces of models, controllers, and views.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout optimize-active-record-queries`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. run `bin/rails rake`,
1. run `bin/rails brakeman`,
1. run `bin/rails rubocop`,
1. run `bin/rails server`,
1. load up http://localhost:3000 in your Web browser of choice and observe that everything works as it always has!

### Before

before

### After

after

### CODE REVIEW

File: app/controllers/tutorials_controller.rb [hunk 1]
```diff
@@ -1,19 +1,9 @@
...
-
-  private
-
-  def tutorial
-    @tutorial ||= Tutorial.find(params[:id])
-  end
-
-  def tutorials
-    @tutorials ||= Tutorial.all
+    @tutorial = Tutorial.includes(:tutorial_steps).find(params[:id])
```
Comment (aileendds) [inline app/controllers/tutorials_controller.rb]: does this execute the find first? e.g.

select * from tutorials where id = <:id>;
select * from tutorial_steps where tutorial_id = <:id>;

File: app/views/service_specific_information/_post.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <header>
+    <%= post.title %>
+    <%- unless post.effective_at.nil? -%>
+      <br><time>Effective <%= post.effective_at.strftime("%B #{post.effective_at.day.ordinalize}, %Y") %></time>
+    <%- end -%>
+  </header>
+  <section>
+    <%= post.content.html_safe %>
+  </section>
+</article>
```
[the remark this answers is not in the document]
Comment (aileendds) [reply app/views/service_specific_information/_post.html.erb]: 💯 
could you pass in the post list instead and add the outer div + for loop to this as well?

File: app/controllers/tutorials_controller.rb (hunk 1 above)
Comment (jgarber623-gov) [reply app/controllers/tutorials_controller.rb]: Yep! The "after" screenshot in the PR description shows that behavior.

File: app/views/service_specific_information/_post.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgarber623-gov) [reply app/views/service_specific_information/_post.html.erb]: I'd _really_ like to do each of the services (`<div class="usa-grid-full tabs-panel" id="…">`) as their own partial, but that gets a bit complicated with the differences in information in the sidebar.

**Pull request merged**